miraclelinux/meta-debian-extended
# systemd: Update from 241-7~deb10u10 to 241-7~deb10u11

# Purpose of pull request

This version fixes following CVEs.

- CVE-2023-7008
- CVE-2023-50387
- CVE-2023-50868

ELA: ELA-1165-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

# How to test the package
1. Build qemuarm64 image
1. Run qemu image
1. Check systemd

## Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
MACHINE = "qemuarm64"
DISTRO_FEATURES_append = " systemd"
DISTRO_FEATURES_BACKFILL_CONSIDERED += "sysvinit"
VIRTUAL-RUNTIME_init_manager = "systemd"
VIRTUAL-RUNTIME_initscripts = "systemd-compat-units"
```

```sh
$ bitbake core-image-minimal
```

## Run qemu image
```sh
$ runqemu qemuarm64 nographic
```
## Check systemd
Check init program is systemd and SysVinit scripts does not exist.
```
root@qemuarm64:~# ls -la /sbin/init 
root@qemuarm64:~# ps | grep systemd
root@qemuarm64:~# ls -F /etc/
```

# Test result
## Build package
Build succeeded.

```sh
$ bitbake systemd
Parsing recipes: 100% |#####################################################| Time: 0:00:31
Parsing of 1359 .bb files complete (0 cached, 1359 parsed). 2381 targets, 83 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-20.04"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.9"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:4a0b104f9088b6ad2f68dd19c52d4c9b452dfe41"
meta-debian-extended = "HEAD:234b9eeb0f71f4f7ba60d73bf2f5015d67aab9d8"
meta-emlinux         = "HEAD:68f46ad75b8633f71e6cfa9e9bceb3db622cfa12"
meta-emlinux-private = "HEAD:3534c7782ba150055729db6720e5c10264c2d0e7"

Initialising tasks: 100% |##################################################| Time: 0:00:00
Sstate summary: Wanted 431 Found 0 Missed 431 Current 0 (0% match, 0% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2009 tasks of which 0 didn't need to be rerun and all succeeded.
$
```
## Package test
The init program is running systemd.
The SysVinit scripts does not exist.

```sh
root@qemuarm64:~# ls -la /sbin/init
lrwxrwxrwx    1 root     root            22 Sep  6 04:49 /sbin/init -> ../lib/systemd/systemd
root@qemuarm64:~#
root@qemuarm64:~# ps | grep systemd
    1 root      9056 S    {systemd} /sbin/init
  108 root     15660 S    /lib/systemd/systemd-journald
  128 systemd- 81488 S    /lib/systemd/systemd-timesyncd
  135 root     13436 S    /lib/systemd/systemd-udevd
  136 systemd-  7860 S    /lib/systemd/systemd-networkd
  151 messageb  4328 S    /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopi
  153 root      7372 S    /lib/systemd/systemd-logind
  154 systemd-  7628 S    /lib/systemd/systemd-resolved
  163 root      3220 S    grep systemd
root@qemuarm64:~#
root@qemuarm64:~# ls -F /etc
busybox.links.nosuid  issue                 nsswitch.conf         skel/
busybox.links.suid    issue.net             os-release@           sysctl.d/
dbus-1/               ld.so.cache           pam.d/                syslog.conf
default/              ld.so.conf            passwd                systemd/
depmod.d/             limits                profile               timestamp
fstab                 login.access          protocols             tmpfiles.d/
group                 login.defs            resolv-conf.systemd@  udev/
group-                machine-id            resolv.conf@          udhcpc.d/
gshadow               modprobe.d/           rpc                   version
gshadow-              modules-load.d/       securetty             xattr.conf
host.conf             motd                  services              xdg/
hostname              mtab@                 shadow
hosts                 networks              shells
root@qemuarm64:~#
```
